lib/outputmanager.js

```diff
@@ -653,14 +654,21 @@ OutputPlugin.prototype.writeCourseAssets = function(tenantId, courseId, destinat
                   // TODO -- This global replace is intended as a temporary solution
                   var replaceRegex = new RegExp("course/assets/" + asset.filename, 'gi');
 
-                  jsonObject['component'] = JSON.parse(JSON.stringify(jsonObject['component']).replace(replaceRegex, "course/" + jsonObject['config']._defaultLanguage + "/assets/" + encodeURIComponent(asset.filename)));
-                  jsonObject['course'] = JSON.parse(JSON.stringify(jsonObject['course']).replace(replaceRegex, "course/" + jsonObject['config']._defaultLanguage + "/assets/" + encodeURIComponent(asset.filename)));
+                  var lang = jsonObject['config']._defaultLanguage;
+                  var newAssetPath = "course/" + lang + "/assets/" + encodeURIComponent(asset.filename);
+
+                  var objects = ['config', 'course', 'article', 'block', 'component'];
```

Comment:
Set up to do the replace for:
- config
- course
- article
- block
- component